### PR TITLE
[EV-3914] Set the controller manager cache RESTMapper back to the DynamicRESTMapper (default changed on k8s upgrade)

### PR DESCRIPTION
**Pull request by Brian-McM**
The controller runtime adding this https://github.com/kubernetes-sigs/controller-runtime/commit/b38c4a526b4cf9ffbcaebd0fe9363d9d64182dd2 which changes how the cache builder and the controller cache options are combined.

Previously if the cache builders REST Mapper was nil, it would default to the controllers rest mapper, which defaults to the DynamicRESTMapper. This rest mapper is not a static mapper, it looks up types that were previously missing when consulted, like the Tier or Licence types.

With the commit mentioned, if the Mapper in the cache builder config is nil that code defaults it to the DiscoveryRESTMapper, which does not look up types that were previously missing. We use the cache builder here https://github.com/tigera/operator/blob/master/main.go#L232 to do some special things for tiers, so the upgrade from controller runtime 0.13 to 0.14 switched the mapper. This is why removing the custom cache fixes the issue, since it means the default is changed back to the DynamicRestMapper.

This should probably be fixed upstream, but for now we can just set the Mapper in the builder to the DynamicRESTMapper as I've done in this commit.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -201,6 +202,12 @@ func main() {
 		os.Exit(0)
 	}
 
+	mapper, err := apiutil.NewDynamicRESTMapper(cfg)
+	if err != nil {
+		log.Error(err, "Failed stuff.")
```
Comment (rene-dekker) [inline main.go:207]: Maybe this needs some polishing : )

**Review by rene-dekker — APPROVED** [review]
lgtm

File: main.go (hunk 1 above)
Comment (Brian-McM) [reply main.go:207]: Haha! I disagree :smile:

**Review by tmjd — APPROVED** [review]
LGTM, the real test for me will be if downstream CI stays clean, but it certainly seems very reasonable that the RESTMapper change was the issue.

**Pull request merged**